src/translator/byte_array_util.cpp

```diff
@@ -5,6 +5,75 @@
 namespace marian {
 namespace bergamot {
 
+// This is a basic validator that checks if the file has not been truncated
+// it basically loads up the header and checks
+
+// This struct and the getter are copied from the marian source, because it's located
+// inside src/common/binary.cpp:15 and we can't include it.
+struct Header {
+  uint64_t nameLength;
+  uint64_t type;
+  uint64_t shapeLength;
+  uint64_t dataLength;
+};
+
+// cast current void pointer to T pointer and move forward by num elements
+template <typename T>
+const T* get(const void*& current, uint64_t num = 1) {
+  const T* ptr = (const T*)current;
+  current = (const T*)current + num;
+  return ptr;
+}
+
+bool validateBinaryModel(AlignedMemory& model, uint64_t fileSize) {
+  const void * current = &model[0];
+  uint64_t memoryNeeded = sizeof(uint64_t)*2; // We keep track of how much memory we would need if we have a complete file
```

Comment:
wrap lines with comments to 80 characters is standard? My config overwrites this autoformat and I always end up bearing the brunt.

Reply:
Dude, we're not 1998, we have 4k monitors. Just. No. All modern projects (including marian) have abandoned that.

Reply:
I'm ok with longer lines.